linked_list/LinkedList.java

```diff
@@ -0,0 +1,217 @@
+/*
+* Following is the code for Linked List
+* T defines the type of Linked List to be created
+*/
+
+import java.util.NoSuchElementException;
+
+// Class to create a Node that contains data and pointer to next Node 
+class Node<T> {
+    T data;     // Data of type T
+    Node<T> next;     // Pointer to next Node
+
+    // Constructor to create a Node
+    public Node(T input) {  
+        this.data = input;
+        this.next = null;
+    }
+}
+
+// Class containing methods to be performed on Linked List
+class LinkedListMethods<T> {
```

Comment:
Can you rename this to `LinkedList_` . This will be used only once in the code so I don't think it will even bother someone who hates trailing underscores. Also we would like to keep the name as close to the correct name (`LinkedList`) as possible so that if a user wants to use it in their code, they will have to do minimal effort.